query-engine/connector-test-kit/src/test/scala/writes/ids/IntIdUpdateSpec.scala

```diff
@@ -0,0 +1,201 @@
+
+    // Check
+    val result = server.query(
+      """
+        |mutation {
+        |  updateTodo(where: {id: 1}, data: {title: "the title"}){
+        |    id
+        |    title
+        |  }
+        |}
+      """.stripMargin,
+      project
+    )
+
+    result.pathAsString("data.updateTodo.title") should equal("the title")
+    result.pathAsLong("data.updateTodo.id") should equal(1)
+  }
+
+  "Updating the id of an item with an id field of type Int with autoincrement" should "error" in {
+    val project = ProjectDsl.fromString {
+      s"""
+         |model Todo {
+         |  id    Int @id @default(autoincrement())
+         |  title String
+         |}
+       """.stripMargin
+    }
+    database.setup(project)
+
+    // Setup
+    val res = server.query(
+      s"""mutation {
+         |  createTodo(data: {title: "initial"}) {title, id}
+         |}""",
+      project = project
+    )
+
+    res.toString should be(s"""{"data":{"createTodo":{"title":"initial","id":1}}}""")
+
+    // Check
+    server.queryThatMustFail(
+      """
+        |mutation {
+        |  updateTodo(where: {id: 1}, data: { title: "the title", id: 2}){
+        |    id
+        |    title
+        |  }
+        |}
+      """.stripMargin,
+      project,
+      errorCode = 2009,
+      errorContains = "Failed to validate the query `Error occurred during query validation & transformation:\\nMutation (object)\\n  ↳ updateTodo (field)\\n    ↳ data (argument)\\n      ↳ TodoUpdateInput (object)\\n        ↳ id (field)\\n          ↳ Field does not exist on enclosing type.` at `.Mutation.updateTodo.data.TodoUpdateInput.id"
+    )
+  }
+
+/*
+  "Updating a field of type Int with autoincrement" should "error" in {
+    val project = ProjectDsl.fromString {
+      s"""
+         |model Todo {
+         |  id         String @id
+         |  counter    Int @default(autoincrement())
+         |  title      String
+         |}
+       """.stripMargin
+    }
+    database.setup(project)
+
+    // Setup
+    val res = server.query(
+      s"""mutation {
+         |  createTodo(data: {id: "the-id", title: "initial"}) {title, id, counter}
+         |}""",
+      project = project
+    )
+
+    res.toString should be(s"""{"data":{"createTodo":{"title":"initial","id":"the-id","counter":1}}}""")
+
+    // Check
+    server.queryThatMustFail(
+      """
+        |mutation {
+        |  updateTodo(where: {id: "the-id"}, data: { counter: 7}){
+        |    id
+        |    title
+        |    counter
+        |  }
+        |}
+      """.stripMargin,
+      project,
+      errorCode = 2009,
+      errorContains = "Failed to validate the query `Error occurred during query validation & transformation:\\nMutation (object)\\n  ↳ updateTodo (field)\\n    ↳ data (argument)\\n      ↳ TodoUpdateInput (object)\\n        ↳ id (field)\\n          ↳ Field does not exist on enclosing type.` at `.Mutation.updateTodo.data.TodoUpdateInput.id"
+    )
+  }
+*/
```

Comment:
Question: should this restriction also apply to non-id auto-increment integers? The issue doesn't mention that this applies only to `@id` fields, but the implementation for creates only applies to ids.

Reply:
(I'll delete or uncomment that test depending on the answer of course)

Reply:
If the main issue is uniqueness, then the answer is: depends. I believe it would be fine to allow updates / inserts if the field is non-unique.

Reply:
I don't think this should error. The motivation was that a user should not be able to update it in order to not run into situations where creation of records becomes impossible.

Reply:
In other words: The main issue is uniqueness, so updating non-unique fields with autoinc is fine.

Reply:
Ok I'm going to extend this to `@unique @default(autoincrement())` then